nomad/structs/workload_id.go

```diff
@@ -222,10 +222,24 @@ type AllocIdentitiesResponse struct {
 	QueryMeta
 }
 
+type WorkloadType int
+
+const (
+	TaskWorkload WorkloadType = iota
+	ServiceWorkload
```

Comment:
```suggestion
	WorkloadTypeTask WorkloadType = iota
	WorkloadTypeService
```

Just me being annoying again, feel free to ignore 😅 

When reading in isolation `TaskWorkload` and `ServiceWorkload` sound like they refer to the workload of a task/service.